vyper/builtin_functions/functions.py

```diff
@@ -857,22 +902,39 @@ def _storage_element_getter(index):
 class Extract32(_SimpleBuiltinFunction):
 
     _id = "extract32"
-    _inputs = [("b", BytesArrayPrimitive()), ("start", SignedIntegerAbstractType())]
-    _kwargs = {"output_type": Optional("name_literal", "bytes32")}
+    _inputs = [("b", BytesArrayPrimitive()), ("start", UnsignedIntegerAbstractType())]
+    # "name_literal" is a placeholder value for TypeTypeDefinition
+    _kwargs = {"output_type": KwargSettings("name_literal", "bytes32")}
```

Comment:
isn't `"name_literal"` the old way of doing things? also -- i'm confused that this works because there is no code to handle `name_literal` downstream